esp-wifi/src/common_adapter/common_adapter_esp32c3.rs

```diff
@@ -61,7 +63,7 @@ pub(crate) fn phy_mem_init() {
 }
 
 pub(crate) unsafe fn phy_enable() {
-    let count = PHY_ACCESS_REF.fetch_add(1, atomic_polyfill::Ordering::SeqCst);
+    let count = PHY_ACCESS_REF.fetch_add(1, Ordering::SeqCst);
```

Comment:
I'm slightly conflicted with these. If atomic-polyfill falls back to critical sections, this would be probably better as a mutex inside the inner critical section. If not, an atomic fetch-add is better. Since this is a target-specific file, maybe we could split the implementation based on this, but that will make the code inconsistent and will cause pain in the future.

Reply:
I think until portable atomic can serve our purpose, we should stick with atomic_polyfill. Doing so allows us to keep the same looking code for all targets, and only targets that don't support atomic cas will pay the CS price.